portfolio/src/main/webapp/script.js

```diff
@@ -81,32 +82,37 @@ function openLightbox(picIndex) {
 
 // Helper function to show specific picture in lightbox
 function showPic(picIndex) {
-  const picsArray = document.getElementsByClassName("lightbox-pic");
-
-  if (picIndex > picsArray.length) {
-    picIndex = 1;	
-  }
-  
-  else if (picIndex < 1) {
-  	picIndex = picsArray.length;
-  }
 
   for (var i = 0; i < picsArray.length; i++) {
       picsArray[i].style.display = "none";
   }
-  
-  picsArray[picIndex - 1].style.display = "block";
-  
+
+  picsArray[picIndex].style.display = "block";
   currPic = picIndex;
+
 }
 
 // Previous and Next picture functionality
 function changePic(byIndex) {
     currPic += byIndex;
+    currPic = (currPic + picsArray.length) % picsArray.length;
```

Comment:
Adding `picsArray.length` to `currPic` is redundant when taking the remainder of `picsArray.length`. (It's like multiplying by one or adding zero.) So you can just say `currPic %= picsArray.length`.

Reply:
I tried that at first, but when trying to chagePic from index 0 to 11, it computed -1 % 12 and outputted -1. I think % acts differently in JS, but let me know if I'm doing something wrong or if there's a better way to do this

Reply:
Ah, yes, my bad. What you had originally is good; adding the length before taking the remainder is the way to go.